hooks/useBalanceInfo.js

```diff
@@ -71,8 +70,15 @@ function useBalanceInfo() {
       const accountAssets = await algodex.http.indexer.indexer
         .lookupAccountAssets(activeWalletObj?.address)
         .do()
-      const assetOptedIn = await accountAssets.assets?.some((asset) => asset['asset-id'] == assetId)
-      setOptedIn(assetOptedIn)
+
+      if (
+        accountAssets.assets?.some((asset) => asset['asset-id'] === assetId && asset['amount'] > 0)
+      ) {
+        return setOptedIn('received')
+      }
+      if (accountAssets.assets?.some((asset) => asset['asset-id'] === assetId)) {
+        return setOptedIn(true)
+      }
```

Comment:
I think it is important to have this type of check but we need to go about it differently. The following logic allows for an unwanted edgecase: What happens when a user sends their vote token balance to the wallet? If they refresh the page the logic will switch back from `received` to `true` since they will no longer have a balance but are still opted in to the asset which allows the person to get extra vote tokens. 

What you should do instead is check the user's transaction history to see if they made any transfer with the assetID. If they did then it is `received`. If not then you check for the opted in status